hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java

```diff
@@ -214,17 +219,29 @@ public void updateLocationInfoList(List<OmKeyLocationInfo> locationInfoList,
       updatedBlockLocations =
           verifyAndGetKeyLocations(locationInfoList, keyLocationInfoGroup);
     }
-    // Updates the latest locationList in the latest version only with
-    // given locationInfoList here.
-    // TODO : The original allocated list and the updated list here may vary
-    // as the containers on the Datanode on which the blocks were pre allocated
-    // might get closed. The diff of blocks between these two lists here
-    // need to be garbage collected in case the ozone client dies.
+
+    // Every time the key commits, the uncommitted blocks should be detected
+    // If client not commit, the blocks should be cleaned by Open Key CleanUp
+    // Service.
+    // keyLocationInfoGroup has the allocated block location info
+    List<OmKeyLocationInfo> uncommittedBlocks =
+        new ArrayList<>(keyLocationInfoGroup.getBlocksLatestVersionOnly());
+    // Only check ContainerBlockID here to avoid the mismatch of the pipeline
+    // field and BcsId in the OmKeyLocationInfo, as the OmKeyInfoCodec ignores
+    // the pipeline field by default and bcsId would be updated in Ratis mode.
+    List<ContainerBlockID> updatedBlockIDs = updatedBlockLocations.stream().
```

Comment:
It's better to use `Set<ContainerBlockID> updatedBlockIDs` here.

Reply:
> It's better to use `Set<ContainerBlockID> updatedBlockIDs` here.

Thx for the suggestion, but it should be fine here.
1. The duplication of BlockIDs is not harmful since it will be only used for checking if two lists have intersections.
2. Committing the duplicate blocks in one request could nearly happen and in **verifyAndGetKeyLocations** the duplicate should be already removed.

Reply:
`Set.contains()` is faster if block count is large.